### PR TITLE
Replaced clamp with std::clamp

### PR DESCRIPTION
**Pull request by HelloShiv**
* [x] Has this change been discussed on [the forum](https://github.com/SFML/SFML/issues/2313) or in an issue before?
* [x] Does the code follow the SFML [Code Style Guide](https://www.sfml-dev.org/style.php)?
* [ ] Have you provided some example/test code for your changes?
* [ ] If you have additional steps which need to be performed list them as tasks!

----

## Description
Changed clamp with std::clamp removed the declaration of clamp declaration from Vulkan.cpp

This PR is related to the issue #2313 

## Tasks

* [x] Tested on Linux
* [ ] Tested on Windows
* [ ] Tested on macOS
* [ ] Tested on iOS
* [ ] Tested on Android

### CODE REVIEW

**Comment by ChrisThrasher** [comment]
Could you please fix all four files I mentioned in that issue?

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Comment by HelloShiv** [comment]
I actually checked 

- Island.cpp 
- Joystick.cpp 
- SFOpenGLView+mouse.mm

But did not found out std::min and std::max used to replace std::clang , mentioning the code parts below

Island.cpp Includes these 
-------

-  elevation      = std::min(std::max(elevation, 0.0f), 1.0f);
- float factor = std::min((elevation - 0.4f) / 0.1f, 1.f);
- float factor = std::min((elevation - snowcapHeight) / 0.05f, 1.f);
- unsigned int rowEnd   = std::min(rowStart + rowBlockSize, resolutionY);

- std::min(std::max(elevation, 0.0f), 1.0f);

------
### In Joystick.cpp 
there is no use of std::min , std::max or self defined clang

------
In SFOpenGLView+mouse.mm
-------
- rawPos.x       = std::min(std::max(origin.x, rawPos.x), origin.x + size.width - 1);
- rawPos.y       = std::min(std::max(origin.y + 1, rawPos.y), origin.y + size.height);

**Comment by ChrisThrasher** [comment]
https://github.com/SFML/SFML/blob/50d86e4755cfb0fecc191bd9d5e226f6a7ebf18a/examples/island/Island.cpp#L310

https://github.com/SFML/SFML/blob/50d86e4755cfb0fecc191bd9d5e226f6a7ebf18a/examples/joystick/Joystick.cpp#L213

https://github.com/SFML/SFML/blob/50d86e4755cfb0fecc191bd9d5e226f6a7ebf18a/src/SFML/Window/OSX/SFOpenGLView%2Bmouse.mm#L387-L388

File: examples/vulkan/Vulkan.cpp [hunk 1]
```diff
@@ -171,12 +172,6 @@ void matrixPerspective(Matrix& result, sf::Angle fov, float aspect, float nearPl
     result[3][3] = 0.f;
 }
 
-// Clamp a value between low and high values
-template <typename T>
-T clamp(T value, T low, T high)
-{
-    return (value <= low) ? low : ((value >= high) ? high : value);
-}
```
Comment (ChrisThrasher) [inline examples/vulkan/Vulkan.cpp:175]:
```suggestion
```
Minor tweak to avoid adding extra empty newlines

**Review by ChrisThrasher — CHANGES_REQUESTED** [review]
Just one tiny recommendation but I like your latest changes!

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Review by eXpl0it3r — APPROVED** [review]

**Comment by eXpl0it3r** [comment]
SFMLBuildMaster: Build this please.

**Review by ChrisThrasher — APPROVED** [review]
Looks good! Let's just be sure to squash merge this one.

File: src/SFML/Window/OSX/SFOpenGLView+mouse.mm [hunk 2]
```diff
@@ -384,8 +384,8 @@ - (NSPoint)cursorPositionFromEvent:(NSEvent*)eventOrNil
         NSSize  size   = [self frame].size;
         NSPoint origin = [self frame].origin;
         NSPoint oldPos = rawPos;
-        rawPos.x       = std::min(std::max(origin.x, rawPos.x), origin.x + size.width - 1);
-        rawPos.y       = std::min(std::max(origin.y + 1, rawPos.y), origin.y + size.height);
+        rawPos.x       = std::clamp(origin.x, rawPos.x, origin.x + size.width - 1);
+        rawPos.y       = std::clamp(origin.y + 1, rawPos.y, origin.y + size.height);
```
Comment (kimci86) [inline src/SFML/Window/OSX/SFOpenGLView+mouse.mm:388]: It makes more sense this way:
```suggestion
        rawPos.x       = std::clamp(rawPos.x, origin.x, origin.x + size.width - 1);
        rawPos.y       = std::clamp(rawPos.y, origin.y + 1, origin.y + size.height);
```
In addition, the behavior of clamp is undefined if the lower bound is greater than the upper bound, which could happen here.

**Review by kimci86 — CHANGES_REQUESTED** [review]

File: src/SFML/Window/OSX/SFOpenGLView+mouse.mm (hunk 2 above)
Comment (ChrisThrasher) [reply src/SFML/Window/OSX/SFOpenGLView+mouse.mm:388]: Oh yeah I see how the previous min/max implementation meant you could mix up the ordering. Great catch.

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

File: examples/vulkan/Vulkan.cpp [hunk 3]
```diff
@@ -871,14 +865,14 @@ class VulkanExample
             return;
         }
 
-        swapchainExtent.width  = clamp<std::uint32_t>(window.getSize().x,
-                                                     surfaceCapabilities.minImageExtent.width,
-                                                     surfaceCapabilities.maxImageExtent.width);
-        swapchainExtent.height = clamp<std::uint32_t>(window.getSize().y,
-                                                      surfaceCapabilities.minImageExtent.height,
-                                                      surfaceCapabilities.maxImageExtent.height);
+        swapchainExtent.width  = std::clamp<std::uint32_t>(window.getSize().x,
```
Comment (kimci86) [inline examples/vulkan/Vulkan.cpp:868]: Is the template argument required?

**Review by kimci86 — APPROVED** [review]

File: examples/vulkan/Vulkan.cpp (hunk 3 above)
Comment (ChrisThrasher) [reply examples/vulkan/Vulkan.cpp:868]: Yes it's required unless we change that `2` to a `2u` which I think I'd prefer.

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

File: examples/vulkan/Vulkan.cpp (hunk 3 above)
Comment (ChrisThrasher) [reply examples/vulkan/Vulkan.cpp]: I was able to remove them all.

**Review by eXpl0it3r — APPROVED** [review]

**Comment by ChrisThrasher** [comment]
Thank you!

**Pull request merged**